lib/http.js

```diff
@@ -0,0 +1,32 @@
+const http = require('http')
+const https = require('https')
+
+const getRequest = (url, timeout = 30 * 1000) =>
+  new Promise((resolve, reject) => {
+    const urlObject = new URL(url)
+    const httpRequestLib = urlObject.protocol === 'https:' ? https : http
+    const httpRequest = httpRequestLib.get(urlObject, { timeout }, res => {
+      const rawData = []
+
+      res.setEncoding('utf8')
+      res.on('data', chunk => rawData.push(chunk))
+      res.on('end', () => {
+        if (res.statusCode < 200 || res.statusCode >= 300) {
+          const errorMsg = res.body || res.statusMessage || `Http Error ${res.statusCode}`
+          reject(errorMsg)
+        } else {
+          try {
+            resolve(rawData && JSON.parse(rawData.join('')))
```

Comment:
You're right, great catch! Yeah, I was trying to avoid parsing an empty string, which would throw an error. Update it here: https://github.com/articulate/authentic/pull/30/commits/2b4286c6d07e9c583a8e2e0e38f9ff8e518ed4ab.